### PR TITLE
Fixing gaussian dof/bokeh dof/ssao/fxaa/motion blur to work with dynamic resolution

### PR DESCRIPTION
**Pull request by lukaschod**
# **Please read;**
## **PR Workflow for the Graphics repository:**
* **All PRs must be opened as draft initially**
* Reviewers can be added while the PR is still in draft
* The PR can be marked as “Ready for Review” once the reviewers have confirmed that **no more changes are needed**
* Tests will start automatically after the PR is marked as “Ready for Review”
* **Do not use [skip ci]** - this can break some of our tooling
* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [x] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ]  Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR. If you do add documentation, make sure to add the relevant Graphics Docs team member as a reviewer of the PR. If you are not sure which person to add, see the [Docs team contacts sheet](https://docs.google.com/spreadsheets/d/1rgUWWgwLFEHIQ3Rz-LnK6PAKmbM49DZZ9al4hvnztOo/edit#gid=1058860420).
- [x] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR
Fix for the case 1225467 (Enterprise Support). The fixed isssue is that depth of field shaders depended on _TexelSize property for effect calculation. However the property does not account for dynamic resolution, which resulted in incorrect calculations. With the fix now its passing custom property _SourceSize.
Fix other post processing effects to work with dynamic resolution (fxaa, ssao, motion blur).

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [x] Built a player
- [x] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: Added gfx test 133_DepthOfField_DynamicResolution
image

**Yamato**: https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/universal%252Fbugfix%252Fpp-dynamic-resolution

---
### Comments to reviewers
The fix itself is somewhat incomplete until we will have RTHandles implemented in URP where instead of calculating width and heigh from dynamic resolution buffer, we will directly pass it from handles.

**For QA**:
- Test if nothing regressed with post processing effects.
- Works correctly on mobile.
- Test with dynamic resolution.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/RendererFeatures/ScreenSpaceAmbientOcclusion.cs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 using System;
+using System.Security.Permissions;
```
Comment (ellioman) [inline com.unity.render-pipelines.universal/Runtime/RendererFeatures/ScreenSpaceAmbientOcclusion.cs:2]: I guess we don't need this include? :)

File: com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs [hunk 2]
```diff
@@ -589,14 +589,32 @@ void DoDepthOfField(Camera camera, CommandBuffer cmd, int source, int destinatio
             if (m_DepthOfField.mode.value == DepthOfFieldMode.Gaussian)
                 DoGaussianDepthOfField(camera, cmd, source, destination, pixelRect);
             else if (m_DepthOfField.mode.value == DepthOfFieldMode.Bokeh)
-                DoBokehDepthOfField(cmd, source, destination, pixelRect);
+                DoBokehDepthOfField(camera, cmd, source, destination, pixelRect);
+        }
+
+        void SetTextureSize(Camera camera, CommandBuffer cmd, int nameID, int width, int height)
```
Comment (ellioman) [inline com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs:595]: Is this function being used anywhere?

File: com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs [hunk 3]
```diff
@@ -699,11 +720,12 @@ static float GetMaxBokehRadiusInPixels(float viewportHeight)
             return Mathf.Min(0.05f, kRadiusInPixels / viewportHeight);
         }
 
-        void DoBokehDepthOfField(CommandBuffer cmd, int source, int destination, Rect pixelRect)
+        void DoBokehDepthOfField(Camera camera, CommandBuffer cmd, int source, int destination, Rect pixelRect)
```
Comment (ellioman) [inline com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs:723]: Do we need to add the camera as a parameter?

File: com.unity.render-pipelines.universal/Runtime/PostProcessUtils.cs [hunk 4]
```diff
@@ -1,3 +1,5 @@
+using UnityEditor.WindowsStandalone;
```
Comment (ellioman) [inline com.unity.render-pipelines.universal/Runtime/PostProcessUtils.cs:1]: I guess this is accidental? Will probably break builds having editor includes

File: com.unity.render-pipelines.universal/ShaderLibrary/SSAO.hlsl [hunk 5]
```diff
@@ -17,8 +17,7 @@ SAMPLER(sampler_ScreenSpaceOcclusionTexture);
 // Params
 float4 _BlurOffset;
 float4 _SSAOParams;
-float4 _BaseMap_TexelSize;
-float4 _CameraDepthTexture_TexelSize;
+float4 _SourceSize;
```
Comment (ellioman) [inline com.unity.render-pipelines.universal/ShaderLibrary/SSAO.hlsl:20]: Just wondering, would it make sense to add this to UnityInput.hlsl and add a function to retrieve it in ShaderVariablesFunctions.hlsl ?

File: com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs (hunk 2 above)
Comment (lukaschod) [reply com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs]: Fixed

File: com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs (hunk 3 above)
Comment (lukaschod) [reply com.unity.render-pipelines.universal/Runtime/Passes/PostProcessPass.cs]: Fixed

File: com.unity.render-pipelines.universal/Runtime/PostProcessUtils.cs (hunk 4 above)
Comment (lukaschod) [reply com.unity.render-pipelines.universal/Runtime/PostProcessUtils.cs]: Fixed

File: com.unity.render-pipelines.universal/ShaderLibrary/SSAO.hlsl (hunk 5 above)
Comment (lukaschod) [reply com.unity.render-pipelines.universal/ShaderLibrary/SSAO.hlsl:20]: I think we can revisit this once RTHandle will be implemented and If @phi-lira decides to add _ScreenSize property in functions form, that we talked about int one of the threads. Now I think there is no point to standardize it as this is not complete solution.

Comment (ellioman) [reply com.unity.render-pipelines.universal/ShaderLibrary/SSAO.hlsl:20]: Sounds good 👍

**Comment by lukaschod** [comment]
Currently blocked until ShaderGraph issue gets fixed https://unity.slack.com/archives/C3KQ4UD6V/p1597244076326500.

**Review by aleksandrasdzikia — APPROVED** [review]
Post processing to be working as expected, no issues or artefacts. No issues on mobile devices found. Tested with a dynamic resolution and without.

**Comment by ellioman** [comment]
Yamato was green before I fixed a changelog conflict.

**Comment by lukaschod** [comment]
Checked that pr is not really mergable to 8.x.x

**Pull request merged**